luavm/src/lib/lua_values/mod.rs

```diff
@@ -155,6 +185,52 @@ impl LuaVal {
     }
 }
 
+impl PartialEq for LuaVal {
+    fn eq(&self, other: &LuaVal) -> bool {
+        if self.is_number() && other.is_number() {
+            return self.to_float().unwrap() == other.to_float().unwrap();
+        } else if self.is_string() && other.is_string() {
+            return self.get_string_ref().unwrap() == other.get_string_ref().unwrap();
+        } else if self.kind() == other.kind() {
+            if self.kind() == LuaValKind::NIL {
+                return true;
+            } else if self.kind() == LuaValKind::TABLE {
+                return unsafe { (*table_ptr(self.val)).same_ptr(&*table_ptr(other.val)) };
+            }
+        }
+        false
+    }
+}
+
+impl Eq for LuaVal {}
+
+impl Hash for LuaVal {
+    fn hash<H: Hasher>(&self, state: &mut H) {
+        match self.kind() {
+            LuaValKind::INT | LuaValKind::FLOAT | LuaValKind::NIL | LuaValKind::TABLE => {
+                // because val is tagged, this means that an int's hash will never collide
```

Comment:
I think this comment is unneeded (and inaccurate, because clearly some int hashes will collide with some table hashes).